pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -243,6 +227,20 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 		r.updateStatus(r.client, namespacedName, cephv1.ConditionReady)
 		return reconcile.Result{}, nil
 	}
+
+	// If no image is specified we don't perform any checks,
+	// for external clusters we don't need to check the version
+	if cephCluster.Spec.CephVersion.Image != "" {
```

Comment:
It seems unsafe to use `cephCluster.Spec.CephVersion.Image != ""` as a proxy for checking if the cluster is external. The cluster could be internal but improperly specified. I think we should add a check for `cephCluster.Spec.External.Enable` instead.

Reply:
removed, as in external case reconcile will be returned prior